Updated all character classes and character controller

Cdttn8 opened the pull request:
Big updates on this merge.

cdttn99 committed:
Updated a lot

I updated animations Updated Animation Triggers Did QOL on prefabs and Animation. Fixed the inability to fire while shift is pressed and character is not moving.  Made direction faced to be equal to the mouse instead of last movement

cdttn99 committed:
Update characters

Character now properly resembles their model in attacks. Changed script to run off of Warrior Class. Added debugging to see circle when character is selected.

cdttn99 committed:
Update CharacterControl.cs

Character now flashes when taking damage

cdttn99 committed:
Update CharacterControl.cs

Animator now works properly no need for each subclass to override basic attack. Added comments for clarity and added debugger to show off invulnerability.

cdttn99 committed:
Merge branch 'main' into Movement

cdttn99 committed:
Update ArcherClass.cs

cdttn99 committed:
Update Lust.unity

removed temp character

NamesAreHard6600 reviewed (changes requested):
Some things that probably should be fixed before merging this:
1. Increase attack range, especially on up and down, since the character is taller than they are wide. Can you have the origin rotate around the character in some kind of ellipse? I know that's probably complicated. 
2. Clearer attacks. The swing animation is great but I have no idea what my actual range is. I feel like there needs to be some sort of indicator of what the range of the attack is, otherwise the melee character is going to be absolutely no fun at all.

cdttn99 committed:
Updated quite a lot

Worked on mele and mage attack logic. Added animations for mele character.  Added a test scene for prefab editing.

cdttn99 committed:
increased range

Increased range

cdttn99 committed:
Merge branch 'main' into Movement

Cdttn8 commented:
Increase attack range, especially on up and down, since the character is taller than they are wide. Can you have the origin rotate around the character in some kind of ellipse? I know that's probably complicated.

    It now does a half circle around the direction the character is facing and is taller and shorter and wider.

    Clearer attacks. The swing animation is great but I have no idea what my actual range is. I feel like there needs to be some sort of indicator of what the range of the attack is, otherwise the melee character is going to be absolutely no fun at all.

I have looked it needs someone in the animation department not the programmer

cdttn99 committed:
Updated names

updated animation names to be more inline

NamesAreHard6600 reviewed (changes requested):
Still some things to work on here. 
1. When the character is dead the hitbox is stays the same. 
2. Hitting enemies up and down still doesn't work super great. 
3. I really don't want to merge this until the attacks are actually visible. Make someone else do it if you don't want to.

cdttn99 committed:
upgraded attack

a

cdttn99 committed:
Update WarriorClass.cs

yeah

cdttn99 committed:
fixed camera issues

camera changes

cdttn99 committed:
Merge branch 'main' into Movement

cdttn99 committed:
Merge branch 'main' into Movement

cdttn99 committed:
Updated

Added hp bars to enemy

cdttn99 committed:
Merge branch 'main' into Movement

cdttn99 committed:
Merge branch 'main' into Movement

cdttn99 committed:
Revert "Merge branch 'main' into Movement"

This reverts commit 8c4d39bebfeed7c5fd2b5113778e6fbe5016f0be, reversing
changes made to 3e15e4fd08b48ce6047a5edf3a8d1477ab450829.

NamesAreHard6600 commented:
Tested and merged in #39 
This branch is a dead branch, due to merging errors.
